test/e2e/actions/helm_related.go

```diff
@@ -25,11 +24,12 @@ func HelmDefaultUpgradeResources(data *model.TestDataProvider) {
 		data.Resources.Users[0].DeleteAllRoles()
 		data.Resources.Users[0].AddBuildInAdminRole()
 		data.Resources.Users[0].Spec.Project.Name = data.Resources.GetAtlasProjectFullKubeName()
-		generation, _ := strconv.Atoi(kubecli.GetGeneration(data.Resources.Namespace, data.Resources.Deployments[0].GetDeploymentNameResource()))
+		generation, err := kubecli.GetDeploymentObservedGeneration(data.Context, data.K8SClient, data.Resources.Namespace, data.Resources.Deployments[0].ObjectMeta.GetName())
+		Expect(err).NotTo(HaveOccurred())
 		helm.UpgradeAtlasDeploymentChartDev(data.Resources)
 
 		By("Wait project creation", func() {
-			WaitDeployment(data.Resources, strconv.Itoa(generation+1))
+			WaitDeployment(data, generation+1)
```

Comment:
I see that we get and assign `generation` just to pass the +1 thing. Potentially can create a `func getNextGen() int` which will get the generation and return the next one. Just an idea 💡

Reply:
I think that we don't need this function at all. This function is needed for tests that I have not refactored.

Reply:
Ok 👌